### ISSUE TITLE
Actions can render offscreen in landscape mode

### ISSUE THREAD

**Issue by petedoyle** [JakeWharton/ActionBarSherlock#339]
I have some actions that have somewhat long text descriptions.  It seems to cause actions to get pushed off-screen (instead of overflowing into the pop-up menu).  

Screenshot 1 - Missing the "Stats" action, which should be right after "Edit Details" (and is not in the popup menu):
http://cl.ly/2m103w0M2F0Z0T1m2B3q

Screenshot 2 - Renamed "Edit Details" to "Edit" so there's space for the "Stats" action (though its text is still offscreen):
http://cl.ly/190h0l0q0s1u3X03290Z

This is with 4.0.0 "RC2" (commit bd157f845)

**Comment by petedoyle** [comment]
On an original Droid, Android 2.3.7

**Comment by JakeWharton** [comment]
Any idea what happens on ICS at the same resolution?

**Comment by petedoyle** [comment]
Checking it out right now.  I personally wouldn't consider it a blocker for 4.0.

**Comment by JakeWharton** [comment]
It may do the same on ICS due the length of the action item text. In which case I'd put smaller text in your `strings.xml` and put the full text in `values-large/strings.xml` or something.

But post an ICS screenshot if you can so we can see....

**Comment by petedoyle** [comment]
Sorry, its at a different resolution (1280x720 instead of 854x480) and the action order is different on ICS for some reason, but it looks like its measuring the text to determine when to overflow:
http://cl.ly/3D3B1z2Q042K1j1A0e2g

I can try to get one at 854x480 on ICS, but it might take a couple hours to boot up a new ICS emulator and sync data for my app.  The emulator is really slow for me.

**Comment by JakeWharton** [comment]
What does your menu XML look like? If you're using a lot of `showAsAction="always"` this would happen. Make sure you're using `ifRoom` as much as possible.

**Comment by petedoyle** [comment]
OK, ICS 854x480:
http://cl.ly/400d1T0R1N3S1R300903

I have an XML file with all actions set to `ifRoom|withText` or `never`.

In addition, Edit Details and Stats are added programmatically with `MenuItem.SHOW_AS_ACTION_ALWAYS | MenuItem.SHOW_AS_ACTION_WITH_TEXT`.  

I can't remember why I did a couple programmatically.  I can check that out.

Also, a couple are set to both hidden and invisible.

**Comment by petedoyle** [comment]
So ICS seems to show the same thing (even pushing the `...` button offscreen).

**Comment by JakeWharton** [comment]
On certain screen sizes only two action items will be displayed on the bar. By using 'always' you force the extra ones which is why they are pushing off the right side.

**Comment by petedoyle** [comment]
Makes sense.  I think I was trying that to ensure those two actions would always be in the actionbar instead of overflowing to the popup menu.  I figured it would just push anything with `ifRoom` to the popup menu, but I can try some other ways to do that (probably just make them first in the XML file with `ifRoom`?).  I just noticed things getting pushed off-screen and thought it might be a bug.

Thanks again for all your crazy hard work, I'm really glad ABS exists.  And from what I can tell 4.0 is really nice (and much cleaner to use) than 3.x.  I was able to just drop in my jar from https://github.com/petedoyle/android-support-v4-googlemaps and go to town.  Really nice.
